cloudmarker/clouds/azsql.py

```diff
@@ -0,0 +1,231 @@
+
+        Yields:
+            tuple: A tuple which when unpacked forms valid arguments for
+                :meth:`_get_server_db_details`.
+
+        """
+        try:
+            tenant = self._tenant
+            creds = self._credentials
+            sub_client = SubscriptionClient(creds)
+            sub_list = sub_client.subscriptions.list()
+
+            for sub_index, sub in enumerate(sub_list):
+                sub = sub.as_dict()
+                _log.info('Found %s', util.outline_az_sub(sub_index,
+                                                          sub, tenant))
+
+                yield from self._get_subscription_dbs(sub_index, sub)
+                # Break after pulling data for self._max_subs number of
+                # subscriptions. Note that if self._max_subs is 0 or less,
+                # then the following condition never evaluates to True.
+                if sub_index + 1 == self._max_subs:
+                    _log.info('Stopping subscriptions fetch due to '
+                              '_max_subs: %d; tenant: %s', self._max_subs,
+                              tenant)
+                    break
+
+        except Exception as e:
+            _log.error('Failed to fetch subscriptions; %s; error: %s: %s',
+                       util.outline_az_sub(sub_index, sub, tenant),
+                       type(e).__name__, e)
+
+    def _get_subscription_dbs(self, sub_index, sub):
+        """Get SQL DBs from a single subscrption.
+
+        Yields:
+            tuple: A tuple which when unpacked forms valid arguments for
+                :meth:`_get_server_db_details`.
+
+        """
+        try:
+            tenant = self._tenant
+            creds = self._credentials
+            sub_id = sub.get('subscription_id')
+            sql_client = SqlManagementClient(creds, sub_id)
+            db_server_list = sql_client.servers.list()
+
+            for server_index, sql_server in enumerate(db_server_list):
+                sql_server = sql_server.as_dict()
+
+                _log.info('Found SQL Server #%d: %s; %s',
+                          server_index, sql_server.get('name'),
+                          util.outline_az_sub(sub_index, sub, tenant))
+                sql_server_id = sql_server.get('id')
+                sql_server_name = sql_server.get('name')
+                rg_name = \
+                    tools.parse_resource_id(sql_server_id)['resource_group']
+                yield (sub, rg_name, sql_server_name,
+                       util.outline_az_sub(sub_index, sub, tenant))
+
+                # Break after pulling data for self._max_recs number
+                # of SQL servers for a subscriber. Note that if
+                # self._max_recs is 0 or less, then the following
+                # condition never evaluates to True.
+                if server_index + 1 == self._max_recs:
+                    _log.info('Stopping get_subscription_dbs fetch due '
```

Comment:
Please rewrite this to `Stopping SQL server fetch due ` to be consistent with `AzVM` as well as with the `Found` and `Failed to fetch` logs in this plugin.